### ISSUE TITLE
Различные доработки запрошенные в оригинальном плагине

### ISSUE THREAD

**Issue by ange007** [ange007/JQueryFormStyler-Modern#13]
Сюда буду скидывать найденные "предложения" на странице оригинального плагина, может что из этого будет полезным.

**Comment by ange007** [comment]
> 1) Уберите из кода инлайн стили и пропишите их в css.
2) В выпадающем блоке добавьте обертывающий див для списка и выставляйте для него min-height. Это нужно для подключения плагина стилизации скролбара.
3) Допишите для ul и li классы по бэму «.jq-selectbox__list» и «.jq-selectbox__item» например.
4) Добавить колбеки на событие инициализации селекта, изменения размера выпадающей часи(на всякий случай), обновления и на destroy.

**Comment by ange007** [comment]
> 1) Добавлял поддержку для телефонов. В текущей версии она есть, но не очень хорошая. Сделайте пожалуйста для Android и iOS чтобы открывался нативный селект(у вас почемуто только для iOS).
2) сделать мультиселект так же выпадающим.(хотябы опцией).
3) Можно сделать опцию для того чтоб при клике на optgroup выбирались все опшены в мультиселекте.

**Comment by ange007** [comment]
>У input number если задать атрибуту «step» десятичное число, то плагин отказывается работать.
По спецификации атрибут step может принимать любое положительное число с плавающей запятой.
Планируется ли поправить это в будущих версиях?

**Comment by ange007** [comment]
>Прикольная фича конечно, красивая, удобная, но есть проблема, не понимает объект в value
например
```javascript
<select>
<option value='{"with": 100,"without": 200, "d": 18, "dr": 16}'> 18
</select>
```
такую конструкцию просто ломает, потом допилю сам

**Comment by pafnuty** [comment]
> 2) В выпадающем блоке добавьте обертывающий див для списка и выставляйте для него min-height. Это нужно для подключения плагина стилизации скролбара.

Стилизация скроллбара навешивается совершенно нормально и без этого, проверено.

**Comment by pafnuty** [comment]
https://github.com/Dimox/jQueryFormStyler/pull/127

**Comment by pafnuty** [comment]
https://github.com/Dimox/jQueryFormStyler/pull/128

**Comment by ange007** [comment]
Приветствую.
Это намёк на перенос? ))

Постараюсь сделать, я пока тут в пределах своего проекта делаю некоторые правки которые мне понадобились _(в частности выношу в CSS скрытие элемента и прочее + вызов триггера onclick для check/radio box)_. А потом уже посмотрю что там можно по переносить.

**Comment by pafnuty** [comment]
Нет, не намёк. Переношу сюда интересные и нужные моменты, что бы при
возможности можно было заняться этим вопросом. Ну а кто заниматься будет -
не столь важно :smile:

14 апр 2017 г. 5:25 PM пользователь "Vladimir B." <notifications@github.com>
написал:

**Comment by yunusga** [comment]
[Dimox/jQueryFormStyler#131](https://github.com/Dimox/jQueryFormStyler/pull/131)

**Comment by yunusga** [comment]
[Dimox/jQueryFormStyler#132](https://github.com/Dimox/jQueryFormStyler/pull/132)

**Comment by ange007** [comment]
Залил обновление.
Надеюсь я ничего сильно не поломал, так как поменял я там много чего.
Боюсь сложновато теперь будет переносить какие-то глобальные правки из оригинального плагина.

**Comment by pafnuty** [comment]
@ange007 круто!
Ну идеи записывать никто же не мешает :)

**Comment by ange007** [comment]
@pafnuty - речь больше о правках непосредственно которые реализует автор оригинального плагина. Но просто та каша которая существовала _(да и сейчас существует)_ - совсем меня не вдохновляет на поддержку. Очень много мусорного кода как по мне.

**Comment by pafnuty** [comment]
Это да.

3 мая 2017 г. 1:22 PM пользователь "Vladimir B." <notifications@github.com>
написал:
